src/main/java/com/dataloom/organization/OrganizationsApi.java

```diff
@@ -1,5 +1,153 @@
+    String EMAIL_DOMAIN_PATH = "/{" + EMAIL_DOMAIN + "}";
+    String PRINCIPALS        = "/principals";
+    String PRINCIPAL_ID      = "pid";
+    String PRINCIPAL_ID_PATH = "/{" + PRINCIPAL_ID + "}";
+    String TYPE              = "type";
+    String TYPE_PATH         = "/{" + TYPE + "}";
+    String ROLES             = "/roles";
+    String MEMBERS           = "/members";
+
+    @GET( ORGANIZATIONS )
+    Iterable<Organization> getOrganizations();
+
+    @POST( ORGANIZATIONS )
+    UUID createOrganizationIfNotExists( @Body Organization organization );
+
+    @GET( ORGANIZATIONS + ID_PATH )
+    Organization getOrganization( @Path( ID ) UUID organizationId );
+
+    @DELETE( ORGANIZATIONS + ID_PATH )
+    Void destroyOrganization( @Path( ID ) UUID organizationId );
+
+    @PUT( ORGANIZATIONS + ID_PATH + TITLE )
+    Void updateTitle( @Path( ID ) UUID organziationId, @Body String title );
+
+    @PUT( ORGANIZATIONS + ID_PATH + DESCRIPTION )
+    Void updateDescription( @Path( ID ) UUID organizationId, @Body String description );
+
+    @GET( ORGANIZATIONS + ID_PATH + EMAIL_DOMAINS )
+    Set<String> getAutoApprovedEmailDomains( @Path( ID ) UUID organizationId );
+
+    @PUT( ORGANIZATIONS + ID_PATH + EMAIL_DOMAINS )
+    Void setAutoApprovedEmailDomain( @Path( ID ) UUID organizationId, @Body Set<String> emailDomain );
+
+    /**
+     * Add multiple e-mail domains to the auto-approval list.
+     *
+     * @param organizationId The id of the organization to modify.
+     * @param emailDomains   The e-mail domain to add to the auto-approval list.
+     */
+    @POST( ORGANIZATIONS + ID_PATH + EMAIL_DOMAINS )
+    Void addAutoApprovedEmailDomains( @Path( ID ) UUID organizationId, @Body Set<String> emailDomains );
+
+    @HTTP( method = "DELETE", hasBody = true, path = ORGANIZATIONS + ID_PATH + EMAIL_DOMAINS )
+    Void removeAutoApprovedEmailDomains( @Path( ID ) UUID organizationId, @Body Set<String> emailDomain );
+
+    /**
+     * Adds a single e-mail domain to the auto-approval list. This will fail for users who are not an owner of the
+     * organization.
+     *
+     * @param organizationId The id of the organization to modify.
+     * @param emailDomain    The e-mail domain to add to the auto-approval list.
+     */
+    @PUT( ORGANIZATIONS + ID_PATH + EMAIL_DOMAINS + EMAIL_DOMAIN_PATH )
+    Void addAutoApprovedEmailDomain( @Path( ID ) UUID organizationId, @Path( EMAIL_DOMAIN ) String emailDomain );
+
+    /**
+     * Removes a single e-mail domain to the auto-approval list. This will fail for users who are not an owner of the
+     * organization.
+     *
+     * @param organizationId The id of the organization to modify.
+     * @param emailDomain    The e-mail domain to add to the auto-approval list.
+     */
+    @DELETE( ORGANIZATIONS + ID_PATH + EMAIL_DOMAINS + EMAIL_DOMAIN_PATH )
+    Void removeAutoApprovedEmailDomain( @Path( ID ) UUID organizationId, @Path( EMAIL_DOMAIN ) String emailDomain );
```

Comment:
you have single and multiple endpoints for adding, so why not have them for removing? on the contrary, does it make sense to have endpoints for adding/removing only a single email, since you can do the same thing with the multiple endpoint?